contrib/lib-autopilot.py

```diff
@@ -273,17 +276,87 @@ def get_candidates(self,k=10):
         
         self.__logger.info("GENERATE CANDIDATES: Found {} nodes with which channel creation is suggested".format(len(candidats)))
         return candidats
+    
+    def __calculate_statistics(self, candidates):
+        """
+        computes statistics of the candidate set about connectivity, wealth and returns a probability density function (pdf) which 
+        encodes which percentage of the funds should be used for each channel with each candidate node
+        
+        the pdf is proportional to the average balance of each candidate and smoothed with a uniform distribtuion currently the
+        smoothing is just a weighted arithmetic mean with a weight of 0.3 for the uniform distribution. 
+        """
+        pdf = {}
+        for candidate in candidates:
+            neighbors = list(self.G.neighbors(candidate))
+            capacity = sum([self.G.get_edge_data(candidate, n)["satoshis"] for n in neighbors])
+            average = capacity / len(neighbors)
+            pdf[candidate] = average
+        cumsum = sum(pdf.values())
+        pdf = {k:v/cumsum for k,v in pdf.items()}
+        w = 0.7
+        print("percentage   smoothed percentage    capacity    numchannels     alias")
+        print("----------------------------------------------------------------------")
+        res_pdf = {}
+        for k,v in pdf.items():
+            neighbors = list(self.G.neighbors(k))
+            capacity = sum([self.G.get_edge_data(k, n)["satoshis"] for n in neighbors])
```

Comment:
Second time we compute this here, might want to annotate the node itself

Reply:
not for now. not sure if current statistics method stays